flask_icu/__init__.py

```diff
@@ -0,0 +1,454 @@
+    that setting is used, or the default from the :attr:`Babel.date_formats`
+    mapping is used) or a format string as documented by Babel.
+
+    This function is also available in the template context as filter
+    named `datetimeformat`.
+    """
+    return _date_format(datetime, rebase, 'datetime', format)
+
+
+def format_date(date=None, format=None, rebase=True):
+    """Return a date formatted according to the given pattern.  If no
+    :class:`~datetime.datetime` or :class:`~datetime.date` object is passed,
+    the current time is assumed.  By default rebasing happens which causes
+    the object to be converted to the users's timezone (as returned by
+    :func:`to_user_timezone`).  This function only formats the date part
+    of a :class:`~datetime.datetime` object.
+
+    The format parameter can either be ``'short'``, ``'medium'``,
+    ``'long'`` or ``'full'`` (in which cause the language's default for
+    that setting is used, or the default from the :attr:`Babel.date_formats`
+    mapping is used) or a format string as documented by Babel.
+
+    This function is also available in the template context as filter
+    named `dateformat`.
+    """
+    return _date_format(date, rebase, 'date', format)
+
+
+def format_time(time=None, format=None, rebase=True):
+    """Return a time formatted according to the given pattern.  If no
+    :class:`~datetime.datetime` object is passed, the current time is
+    assumed.  By default rebasing happens which causes the object to
+    be converted to the users's timezone (as returned by
+    :func:`to_user_timezone`).  This function formats both date and
+    time.
+
+    The format parameter can either be ``'short'``, ``'medium'``,
+    ``'long'`` or ``'full'`` (in which cause the language's default for
+    that setting is used, or the default from the :attr:`Babel.date_formats`
+    mapping is used) or a format string as documented by Babel.
+
+    This function is also available in the template context as filter
+    named `timeformat`.
+    """
+    return _date_format(time, rebase, 'time', format)
+
+
+def _date_format(datetime, rebase, datetime_type, format):
+    """Internal helper that looks up and creates the correct DateFormat
+    object, and then uses it to format the date string and return it.
+    If rebase is set to true, it automatically reformats the time to the
+    user's timezone.
+    """
+    locale = get_locale()
+    icu = _request_ctx_stack.top.app.extensions['icu']
+    if format is None:
+        format = icu.date_formats[datetime_type]
+    if format in ('short', 'medium', 'long', 'full'):
+        tmp = icu.date_formats["%s.%s" % (datetime_type, format)]
+        is_custom = False if tmp is None else True
+        format = tmp if is_custom else icu.icu_date_formats[format]
+    if is_custom:
+        formatter = SimpleDateFormat(format, locale)
+    else:
+        if datetime_type is 'time':
```

Comment:
For checking strings `==` is preferred over `is`, unless you checking against `None`. The difference between them is [well explained here](http://stackoverflow.com/questions/132988/is-there-a-difference-between-and-is-in-python), but it comes down to `is` is an identity check, while `==` check equality (which you can overwrite in python), allowing other objects to equal the same. We should allow other objects to equal the same here.